compiler/src/main/kotlin/visitors/typechecker.kt

```diff
@@ -233,25 +233,60 @@ class TypeChecker(symbolTable: Table) : ScopedASTVisitor(symbolTable = symbolTab
         node.setType(node.arr.getType())
     }
 
-    override fun visit(node: ArrayBodyExpr) {
+    /**
+     * Finds the size of the largest subarray in each dimension.
+     * Note: Note if their is empty array literals in the tree, this function may return fewer dimensions than expected.
```

Comment:
```suggestion
     * Note: if there are empty array literals in the tree, this function may return fewer dimensions than expected.
```

Also, shouldn't we avoid this behaviour?

Reply:
In the case of `[10][20][30]bool{}` searchSize would return a and list of length 1. I don't see how we would avoid this without forbidding empty array literals

Reply:
Oh then I misunderstood. As long as `{}` returns [0] and `{{{}}}` returns [1, 1, 0] even then it's `[10][10][10][10]bool{{{}}}`